src/listpack.c

```diff
@@ -860,34 +912,40 @@ unsigned char *lpInsert(unsigned char *lp, unsigned char *elestr, unsigned char
         dst = lp + poff;
     }
 
-    /* Store the entry. */
+    unsigned char *last = dst; /* Keep pointer to the last added entry */
```

Comment:
why is the last added entry? if so, we can't continue iterator listapck after indertion or deletion.

Reply:
Currently, if you add 'a' to listpack, newp will point to 'a', right?
In the new one, if you add 'a', 'b', it will point 'b'. Isn't that correct? What do you think it should point to?

Reply:
yes, you're right, maybe we should update the fowlloing comment.
`the element` -> `the last element`?
```
 * If 'newp' is not NULL, at the end of a successful call '*newp' will be set
 * to the address of the element just added, so that it will be possible to
 * continue an interaction with lpNext() and lpPrev().
```

Reply:
Ok, sure.

Also, I'm a bit confused as well :) I'll add tests (if missing) for replace, del and add to check newp value